collector/tcpstat.go

```diff
@@ -0,0 +1,151 @@
+// +build !notcpstat
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	procTcpStat  = "/proc/net/tcp"
+	procTcp6Stat = "/proc/net/tcp6"
+)
+
+type TcpConnectionState int
+
+const (
+	TCP_ESTABLISHED TcpConnectionState = iota + 1
+	TCP_SYN_SENT
+	TCP_SYN_RECV
+	TCP_FIN_WAIT1
+	TCP_FIN_WAIT2
+	TCP_TIME_WAIT
+	TCP_CLOSE
+	TCP_CLOSE_WAIT
+	TCP_LAST_ACK
+	TCP_LISTEN
+	TCP_CLOSING
+)
+
+type tcpStatCollector struct {
+	config Config
+	metric *prometheus.GaugeVec
+}
+
+func init() {
+	Factories["tcpstat"] = NewTcpStatCollector
+}
+
+// NewTcpStatCollector takes a config struct and returns
+// a new Collector exposing network stats.
+func NewTcpStatCollector(config Config) (Collector, error) {
```

Comment:
Sorry, should have noticed this before:

`NewTcpStatCollector` -> `NewTCPStatCollector`

(https://github.com/golang/go/wiki/CodeReviewComments#initialisms)